app/src/lib/stores/helpers/fetch-tags-to-push-memoized.ts

```diff
@@ -0,0 +1,77 @@
+import { fetchTagsToPush } from '../../git'
+import { Repository } from '../../../models/repository'
+import { IGitAccount } from '../../../models/git-account'
+import { IRemote } from '../../../models/remote'
+
+const cache = new Map<string, Map<string, ReadonlyArray<string>>>()
```

Comment:
one thing i think @niik would note here is that this cache can grow unbounded. i don't think that's likely to matter in this case though, as these arrays will generally be quite short or empty.

Reply:
on second thought, with the key specificity here, it might be worth being safe by removing the previous entry for a branch or sha when a new one is cached? i don't feel strongly about this and am happy to leave this alone until it becomes a problem, if it ever does.

Reply:
The growing of the cache gets a bit limited since it gets cleared for a repository every time there's a fetch/push/pull, but I agree that it can still grow unbounded on certain scenarios 👍

I'm going to try to solve this issue by using a LRU cache as described in https://github.com/desktop/desktop/pull/9576/files#r416450730.

Reply:
ah i missed the part where the _entire_ cache gets cleared after a push/pull/fetch! i'm much less worried, but i think this is a good pattern for us to use in other places :)